broadlink/climate.py

```diff
@@ -201,3 +201,252 @@ def set_schedule(self, weekday: List[dict], weekend: List[dict]) -> None:
             input_payload.append(int(weekend[i]['temp'] * 2))
 
         self.send_request(input_payload)
+
+class tornado(device):
+    """Controls a Tornado 16X SQ air conditioner."""
+    def __init__(self, *args, **kwargs):
+        device.__init__(self, *args, **kwargs)
+        self.type = "Tornado air conditioner"
+    
+    def _decode(self, response) -> bytes:
+        payload = self.decrypt(bytes(response[0x38:]))
+        return payload
+    
+    def _calculate_checksum(self, packet:list, target:int=0x20017) -> tuple:
+        """Calculate checksum of given array,
+        by adding little endian words and subtracting from target.
+        
+        Args:
+            packet (list/bytearray/bytes): 
+        """
+        result = target - (sum([v if i % 2 == 0 else v << 8 for i, v in enumerate(packet)]) & 0xffff)
+        return (result & 0xff, (result >> 8) & 0xff)
```

Comment:
Are you sure you are not overcomplicating things here? I haven't stopped to look at this logic carefully, but something tells me it could be simpler.